### ISSUE TITLE
Getting wrong types with prisma client extensions

### ISSUE THREAD

**Issue by jakeleventhal** [prisma/prisma#19958]
### Bug description

The findFirst interface doesnt report incorrect types on `4.16.1`. On `4.15` I didn't have this issue.

### How to reproduce

```ts
const x = await prisma.user.findFirst({ stripeCustomerId: customerId });
```

This code is definitely incorrect, it needs a `where` clause. But it doesn't report it as an error.

If I get rid of the extension code, it correctly reports a type error.

### Expected behavior

Type error is shown

### Prisma information

```ts
const client = (url: string | undefined) =>
  new PrismaClient(url ? { datasources: { db: { url } } } : undefined).$extends({
    model: {
      $allModels: {
        async exists<T>(this: T, where: Prisma.Args<T, 'findFirst'>['where']): Promise<boolean> {
          const context = Prisma.getExtensionContext(this);
          const result = await (context as any).count({ where });
          return result > 0;
        }
      }
    }
  });
```

Getting rid of the extension works

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: 18.12.1

### Prisma Version

```
4.16.1
```

**Comment by millsp** [comment]
Hey @jakeleventhal, I believe this has been fixed already in an additional candidate to an upcoming patch. I opened a PR that included a fix for this, would you mind installing trying version `4.17.0-integration-client-extensions-fluent-api.1` ?

**Comment by Jolg42** [comment]
I did a reproduction using the user information and `4.17.0-integration-client-extensions-fluent-api.1` fixed it.

**Comment by jakeleventhal** [comment]
I can verify, this is fixed with `4.17.0-integration-client-extensions-fluent-api.1`

looking forward to next release. thanks for the speedy fix

**Comment by Jolg42** [comment]
:shipit: Note: This fix is now available as [`4.16.2`](https://github.com/prisma/prisma/releases/tag/4.16.2)
